src/lib/OpenEXR/ImfStandardAttributes.h

```diff
@@ -140,24 +155,310 @@ IMF_STD_ATTRIBUTE_DEF (longitude, Longitude, float)
+// If present, the value should contain only printable characters
+// and have a nonzero length.
+//
+
+IMF_STD_ATTRIBUTE_DEF (lensSerialNumber, LensSerialNumber, std::string)
+
+//
+// lensFirmwareVersion -- firmware version of the lens
+//
+// If present, the value should contain only printable characters
+// and have a nonzero length.
+//
+
+IMF_STD_ATTRIBUTE_DEF (lensFirmwareVersion, LensFirmwareVersion, std::string)
+
+//
+// nominalFocalLength -- number printed on barrel of a prime lens, or
+// number next to index mark on a zoom lens, in units of millimeters.
+//
+// Nominal focal length is appropriate for asset tracking of lenses (e.g.
+// a camera rental house catalogs its lens stock by nominal focal length).
+//
+
+IMF_STD_ATTRIBUTE_DEF (nominalFocalLength, NominalFocalLength, float)
+
+//
+// pinholeFocalLength -- in the simplest model of image formation, the
+// distance between the pinhole and the image plane, in units of millimeters.
+//
+// When a CGI application supplies a method for an artist to provide focal
+// length to some calculation, pinhole focal length is almost always the
+// appropriate number to convey to the application.
+//
+
+IMF_STD_ATTRIBUTE_DEF (pinholeFocalLength, PinholeFocalLength, float)
+
+//
+// effectiveFocalLength -- in the thick lens model, the effective focal
+// length is the distance between the front focal point and the front
+// nodal point, or equivalently the back focal point and the back nodal
+// point, in units of millimeters.
+//
+// The effective focal length is an abstraction used in lens design and,
+// unless a CGI application is sophisticated enough to be using the thick
+// lens model, should not be supplied to the application; for normal
+// CGI applications, pinhole focal length should be used.
+//
+// Note that the forward and back lens nodal points mentioned above are
+// distinct in meaning and in position from the forward and back lens
+// entrance pupils. A 'no-parallax' rotation is rotation around the
+// forward lens entrance pupil.
+//
+
+IMF_STD_ATTRIBUTE_DEF (effectiveFocalLength, EffectiveFocalLength, float)
+
+//
+// entrancePupilOffset -- the axial distance from the entrance pupil
+// to the image plane, in units of millimeters. Note that in some 
+// lens configurations, the entrance pupil offset can be negative.
+//
+
+IMF_STD_ATTRIBUTE_DEF (entrancePupilOffset, EntrancePupilOffset, float)
+
+//
+// aperture -- the f-number of the lens, computed as the ratio of
+// lens focal length to diameter of lens entrance pupil at the time
+// the image was created or captured
+// 
+// The lens focal length used in the calculation is the nominal focal
+// length which would be stored in the focalLength attribute, even if
+// the effective focal length (compensating for any focus breathing)
+// is known and stored in an effectiveFocalLength attribute.
 //
 
 IMF_STD_ATTRIBUTE_DEF (aperture, Aperture, float)
 
+//
+// tStop -- ratio of lens focal length to diameter, adjusted for
```

Comment:
Should we also note that for cinema lenses marked about the ring in T numbers, that the tStop is "as read from the dial"?

since we have aperture standing in for fStop, and also tStop, what is the intended resolution of a possible conflict if the fStop and tStop values in a file are not related by a proper mathematical relationship?

Reply:
Oh and should we have a one-way convenience function for turning a floating-point tStop into a string of the form "X +Y/10", i.e. a stop of 5.6 +3/10 ?

Reply:
'diameter of entrance pupil' would assume a perfectly circular aperture, right? Would it be more accurate to say it's the square root of the area of the entrance pupil divided by pi?

Reply:
What I meant by not related by a proper relationship is, what if on the day we recorded f8.0, and on another day someone foolishly scribbled in T33 for unknown reasons. Probably the answer to my question is that exrcheck could flag such a discrepency; outside the scope of this PR.

I'm fine with a convenience function but wouldn't hold it as necessary for this PR.

Reply:
understood as "as recorded on set" per tsc meeting